src/zkp/pimod.rs

```diff
@@ -113,8 +113,15 @@ impl Proof for PiModProof {
 
     #[cfg_attr(feature = "flame_it", flame("PaillierBlumModulusProof"))]
     fn verify(&self, input: &Self::CommonInput) -> Result<()> {
+        // Verify that proof is sound
+        if self.elements.len() != LAMBDA {
+            verify_err!(format!(
```

Comment:
I am! The macro hides the error generation -- it makes an `Err(InternalError::FailedToVerifyProof("PiMod proof is not sound...")`, and then L122 propagates it.

Omar has also identified this something to remove when we refactor our error schemes, because it isn't particularly obvious right now what's happening.

Reply:
Yea I definitely second Omar's comment. My reading of the macro and return type of `verify` was that it was just printing the error. Anyway, cool beans.